internal/torcx/perform.go

```diff
@@ -114,14 +115,23 @@ func applyImages(applyCfg *ApplyConfig, images []Image) error {
 			"reference": im.Reference,
 		}
 
-		tgzArchive, err := storeCache.ArchiveFor(im)
+		archive, err := storeCache.ArchiveFor(im)
 		if err != nil {
 			logrus.WithFields(logFields).Error(err)
 			failedImages = append(failedImages, im)
 			continue
 		}
 
-		imageRoot, err := unpackTgz(applyCfg, tgzArchive.Filepath, im.Name)
+		var imageRoot string
+		switch archive.Format {
+		case ArchiveFormatTgz:
+			imageRoot, err = unpackTgz(applyCfg, archive.Filepath, im.Name)
+		case ArchiveFormatSquashfs:
+			imageRoot, err = mountSquashfs(applyCfg, archive.Filepath, im.Name)
+		default:
+			logrus.WithFields(logFields).Error("unrecognized format for archive: ", archive)
```

Comment:
does it make sense to set `err` here so we track failed images?

Reply:
I think it does make sense, though realistically this code is unreachable so could also panic (since `ArchiveFor` several lines up can only return a tgz or squashfs).

updated to add to failed images so it will fail more noisily if the code changes or I'm missing something.